core/src/main/java/hudson/model/Queue.java

```diff
@@ -864,9 +822,11 @@ private WaitingItem peek() {
      * This method is primarily added to make UI threads run faster.
      *
      * @since 1.483
+     * @deprecated Use {@link #getItems()} directly. As of 1.607 the approximation is no longer needed.
      */
+    @Deprecated
     public List<Item> getApproximateItemsQuickly() {
-        return itemsView.get();
+        return Arrays.asList(getItems());
```

Comment:
That was the first intention: cache using snapshot and authentication as the key (as the snapshot is filtered by the permissions of the logged user, which is the root reason invalid values were being returned, the existing cache was not taking the authentication into account).

After some conversation, only removing the existing cache was considered necessary though adding a correct one is not discarded if needed in the future.